src/frame.jl

```diff
@@ -1,40 +1,38 @@
 # Custom SVD solver for least squares problems
 function solvesvd(A::AbstractArray, b::AbstractVector; tol::Real=1e-7)
-    Am = A[1:end,1:end] # BlockBandedMatrices cannot do SVD
-    U,S,V = svd(Am)
-    S⁺ = inv.(S)[:]
-    S⁺[S⁺ .> 1/tol] .= 0
-    c = V * Diagonal(S⁺) * U' * b
-    return c
+    U,σ,V = svd(Matrix(A)) # BlockBandedMatrices cannot do SVD
+    filter!(>(tol), σ)
+    r = length(σ)
+    return V[:,1:r] * (inv.(σ) .* (U[:,1:r]' * b))
 end
 
 # Affine transformation
 at(a,b,x) = (b-a)/2 * x .+ (b+a)/2
 
 # Construct collocation points
-function collocation_points(M::Int, Me::Int; a::AbstractVector=[-1.,1.], endpoints::AbstractVector=[-5.,5.], innergap::Real = 0.)
-    Tp = eltype(a)
-    el_no = length(a)-1
+function collocation_points(M::Int, Me::Int; I::AbstractVector=[-1.,1.], endpoints::AbstractVector=[-5.,5.], innergap::Real = 0.)
+    Tp = eltype(I)
+    el_no = length(I)-1
 
     x = Array{Tp}(undef,el_no*M+2*Me)
     xnodes = LinRange{Tp}(innergap,1-innergap,M)
     # chebnodes = sort(cos.(π.*xnodes))
 
     xxnodes = LinRange{Tp}(-1+innergap,1-innergap,M)
     for el = 1:el_no
-        x[(el-1)*M+1:el*M] = at(a[el], a[el+1], xxnodes) 
+        x[(el-1)*M+1:el*M] = at(I[el], I[el+1], xxnodes) 
     end
     xnodes = LinRange{Tp}(innergap,1-innergap,Me)
     # chebnodes = sort(cos.(π.*xnodes))
 
     xxnodes = LinRange{Tp}(-1+innergap,1-innergap,Me)
-    x[el_no*M+1:el_no*M+Me] = at(endpoints[1], a[1], xxnodes) 
-    x[el_no*M+1+Me:el_no*M+2*Me] = at(a[end],endpoints[2],xxnodes)
+    x[el_no*M+1:el_no*M+Me] = at(endpoints[1], I[1], xxnodes) 
+    x[el_no*M+1+Me:el_no*M+2*Me] = at(I[end],endpoints[2],xxnodes)
     return sort(unique(x))
 end
 
 # Convert function evaluation to Riemann sum
-function riemann(x::AbstractVector, f::Function)
+function riemann(x::AbstractVector, f::Union{Function, Interpolations.GriddedInterpolation})

```

Comment:
Why not just leave `f` not type restricted? This is only ever needed when you overload for a specific type:
```suggestion
function riemann(x::AbstractVector, f)
```